private/buf/bufsync/syncer.go

```diff
@@ -57,13 +56,15 @@ type syncer struct {
 
 	// commitsToTags holds all tags in the repo associated to its commit hash. (commits:[]tags)
 	commitsToTags map[string][]string
-	// branchesToModulesForSync holds all the branches, module directories, and module identity
-	// targets for those directories, prepared before syncing either from its identity override or
-	// HEAD commit. (branch:moduleDir:targetModuleIdentity)
-	branchesToModulesForSync map[string]map[string]bufmoduleref.ModuleIdentity
-	// modulesToBranchesLastSyncPoints holds remote expected sync points for module identity and its
+	// modulesDirsToBranchesToIdentities holds all the module directories, branches, and module identity
+	// targets for those directories+branches, prepared before syncing either from its identity
+	// override or HEAD commit. (moduleDir:branch:targetModuleIdentity)
+	modulesDirsToBranchesToIdentities map[string]map[string]bufmoduleref.ModuleIdentity
```

Comment:
This represents the main change for this PR, no longer `branch -> moduleDir` but `moduleDir -> branch`